llm_studio/python_configs/text_causal_language_modeling_config.py

```diff
@@ -502,4 +502,9 @@ def check(self) -> Dict[str, List]:
             errors["message"] += [
                 "Please enable do sample if you want to use temperature > 0."
             ]
+        if self.prediction.temperature == 0 and self.prediction.do_sample:
```

Comment:
we might also need the other way around when temp is set and do_sample not

Reply:
Good point, will add it. 
Interestingly, in that case `model.generate` will not fail, but just log a warning (and set T=0).

Reply:
Decided to leave it as is for the moment, probably better solution is to replace `do_sample = T > 0` and then set a warning if `top_k=0`. In any case, settings will be adapted automatically in this case.